pkg/services/live/features/dashboard.go

```diff
@@ -28,16 +45,97 @@ func (h *DashboardHandler) GetHandlerForPath(path string) (models.ChannelHandler
 }
 
 // OnSubscribe for now allows anyone to subscribe to any dashboard
-func (h *DashboardHandler) OnSubscribe(ctx context.Context, _ *models.SignedInUser, e models.SubscribeEvent) (models.SubscribeReply, backend.SubscribeStreamStatus, error) {
-	return models.SubscribeReply{
-		Presence:  true,
-		JoinLeave: true,
-	}, backend.SubscribeStreamStatusOK, nil
+func (h *DashboardHandler) OnSubscribe(ctx context.Context, user *models.SignedInUser, e models.SubscribeEvent) (models.SubscribeReply, backend.SubscribeStreamStatus, error) {
+	parts := strings.Split(e.Path, "/")
+	if parts[0] == "gitops" {
+		// gitops gets all changes for everything, so lets make sure it is an admin user
+		if !user.HasRole(models.ROLE_ADMIN) {
+			return models.SubscribeReply{}, backend.SubscribeStreamStatusPermissionDenied, nil
+		}
+		return models.SubscribeReply{
+			Presence:  true,
+			JoinLeave: true, // ?? likely not necessary
```

Comment:
If we don't need it - then better to not use. The reason is that `join/leave` events are pretty expensive for channels with many subscribers - since on every connect/reconnect they are copied to everyone listening. Maybe this is not a problem here but I'd start without them anyway until really required.

Reply:
removed